core_perception/lidar_localizer/nodes/ndt_matching/ndt_matching.cpp

```diff
@@ -1551,84 +1566,45 @@ int main(int argc, char** argv)
   private_nh.getParam("imu_upside_down", _imu_upside_down);
   private_nh.getParam("imu_topic", _imu_topic);
   private_nh.param<double>("gnss_reinit_fitness", _gnss_reinit_fitness, 500.0);
-  private_nh.getParam("output_tf_frame_id", _output_tf_frame_id);
+  private_nh.getParam("base_frame", _base_frame);
 
-  std::string lidar_frame;
-  nh.param("localizer", lidar_frame, std::string("lidar"));
-  tf2_ros::Buffer tf_buffer;
-  tf2_ros::TransformListener tf_listener(tf_buffer);
-  geometry_msgs::TransformStamped tf_baselink2primarylidar;
-  bool received_tf = true;
 
-  // 1. Try getting base_link -> lidar TF from TF tree
-  try
+  if (nh.getParam("localizer", _localizer) == false)
+  {
+    std::cout << "localizer is not set." << std::endl;
+    return 1;
+  }
+  if (nh.getParam("tf_x", _tf_x) == false)
   {
-    tf_baselink2primarylidar =
-      tf_buffer.lookupTransform("base_link", lidar_frame, ros::Time().now(), ros::Duration(3.0));
+    std::cout << "tf_x is not set." << std::endl;
+    return 1;
   }
-  catch (tf2::TransformException& ex)
+  if (nh.getParam("tf_y", _tf_y) == false)
```

Comment:
Are we loading the lidar transform as a param here instead of from the URDF data? If we're making changes here, is this a good opportunity to fix that issue we ran into regarding the discrepancy between the NDT configured transform and the URDF data by making the NDT read the UDRF?

Reply:
As discussed on slack, we will leave this as is since this is a version revert for this node.